honeybee_radiance_folder/folder.py

```diff
@@ -500,6 +506,66 @@ def dynamic_scene(self, indoor=False, reload=False):
         return self._dynamic_scene_indoor if indoor \
             else self._dynamic_scene
 
+    def grid_info_from_input(self):
+        """Return the list containing information about sensor grids that
+        are written to the folder after model export"""
+
+        infoJson = os.path.join(self.grid_folder(full=True), "_info.json")
+        return parse_grid_info(infoJson, validate=True)
+
+    def grid_info_from_model(self):
+        """Return the list containing information about sensor grids that
+           are extracted from the model after running the simulation."""
+
+        infoJson = os.path.join(self.grid_folder(full=True),
+                                "_model_grids_info.json")
+        return parse_grid_info(infoJson, validate=True)
+
+    def _grid_data_from_info_json(self, info_type):
+        """
+        This is an internal function that for a specified info_json,
+        returns the consolidated grid data ,that includes grid names,counts
+        and identifiers, as a single dictionary.
+        Args:
+            info_type: Can be 'model' or 'input'. If specified as 'model',
+            then the '_info.json' file will be parsed. If specified as 'input'
+            then then the '_model_grids_info.json' will be parsed.
+
+        Returns:
+            A dictionary containing consolidated grid data.
+
+        """
+
+        info_data = {'input': self.grid_info_from_input,
+                     'model': self.grid_info_from_model}[info_type]
+
+        grid_info = info_data()
+
+        grid_folder = self.grid_folder(full=True)
+
+        grid_data_dict = {}
+        if grid_info:
+            for grid in grid_info:
+                grid_id = grid.pop('full_id')
+                grid_data_dict[grid_id] = parse_grid_json(
+                    os.path.join(grid_folder, grid_id + '.json'))
+                grid_data_dict[grid_id].update(grid)
```

Comment:
This is super confusing to me. Why do you need to pop and then update?

Reply:
The pop is because the key of the dictionary is the full_id. Then the grid_json and rest of the data from the popped dictionary are merged into a single dictionary. So, I want to retain this logic as is.

Reply:
OK. Just wanted to make sure. If you think that's the way to go then let's keep it as is.